packages/client/hmi-client/src/services/knowledge.ts

```diff
@@ -1,14 +1,14 @@
-import API, { Poller, PollerState, PollResponse } from '@/api/api';
+import API, { Poller, PollerState, PollResponse, PollerResult } from '@/api/api';
 import { AxiosError, AxiosResponse } from 'axios';
-import { Artifact, Model } from '@/types/Types';
+import { Artifact, ExtractionResponse, Model } from '@/types/Types';
 import { logger } from '@/utils/logger';
 
 /**
  * Fetch information from the extraction service via the Poller utility
  * @param id
  * @return {Promise<PollerResult>}
  */
-export async function fetchExtraction(id: string) {
+export async function fetchExtraction(id: string): Promise<PollerResult<any>> {
```

Comment:
If you want to specifically add the type, please do so, but don't add `any`.

Reply:
It looks like [PollerResult](https://github.com/DARPA-ASKEM/Terarium/blob/fe352e45f229a30f748f381f6b6914e9d5438be9/packages/client/hmi-client/src/api/api.ts#L70-L73) is a generic interface with a placeholder `<T>` type parameter, where `PollerResult.data` is `<T>`. So I think the return type _should_ be `PollerResult<any>`? But also don't have the best understanding when generic types (`T`) are involved. Do you have a recommendation on what the correct return type would be?

Reply:
Poller isn't the easiest thing to use currently because it wants to be agnostic to errors/progress/data but needs to know the types, let's leave it and I can craft better examples or tightening up the API.